node/subsystem-types/src/messages.rs

```diff
@@ -194,11 +194,16 @@ pub enum CollatorProtocolMessage {
 	///
 	/// This should be sent before any `DistributeCollation` message.
 	CollateOn(ParaId),
-	/// Provide a collation to distribute to validators with an optional result sender.
+	/// Provide a collation to distribute in the given active leaf to validators with an optional result sender.
 	///
 	/// The result sender should be informed when at least one parachain validator seconded the collation. It is also
 	/// completely okay to just drop the sender.
-	DistributeCollation(CandidateReceipt, PoV, Option<oneshot::Sender<CollationSecondedSignal>>),
+	DistributeCollation(
+		Hash,
```

Comment:
This leaf hash corresponds to the fragment tree we will try to put a candidate it.
For example, collation generation asynchronously awaits for new collations, once it's ready, sends it along with a leaf-descendant of candidate relay parent.